extensions/BMO/Extension.pm

```diff
@@ -51,6 +51,9 @@ use List::Util qw(first);
 use Scalar::Util qw(blessed);
 use Sys::Syslog qw(:DEFAULT);
 use Text::Balanced qw( extract_bracketed extract_multiple );
+use URI;
+use URI::QueryParam;
+use URI::Split qw(uri_split uri_join);
```

Comment:
you needn't use the URI::Split, the URI object has methods available (I'll explain below)